jormungandr/src/settings/logging.rs

```diff
@@ -215,30 +232,81 @@ impl LogSettingsEntry {
+                guards.push(guard);
+
+                match settings.format {
+                    LogFormat::Default | LogFormat::Plain => {
+                        let layer = tracing_subscriber::fmt::Layer::new()
+                            .with_level(true)
+                            .with_writer(non_blocking);
+                        (Some(layer), None)
+                    }
+                    LogFormat::Json => {
+                        let layer = tracing_subscriber::fmt::Layer::new()
+                            .json()
+                            .with_level(true)
+                            .with_writer(non_blocking);
+                        (None, Some(layer))
+                    }
+                }
+            } else {
+                (None, None)
             }
-            #[cfg(feature = "gelf")]
-            LogOutput::Gelf {
-                backend: address,
-                log_id: _graylog_source,
-            } => {
-                format.require_default()?;
-                // TODO: maybe handle this tasks outside somehow.
+        } else {
+            (None, None)
+        };
+        #[cfg(feature = "systemd")]
+        let journald_layer = if let Some(settings) = layer_settings.journald {
+            settings.format.require_default()?;
+            let layer = tracing_journald::layer().map_err(Error::Journald)?;
+            Some(layer)
+        } else {
+            None
+        };
+        #[cfg(feature = "gelf")]
+        let gelf_layer = if let Some(settings) = layer_settings.gelf {
+            // have to use if let because it's an enum
+            if let LogOutput::Gelf { backend, .. } = settings.output {
                 let (layer, task) = tracing_gelf::Logger::builder()
-                    .connect_tcp(address.clone())
+                    .connect_tcp(backend)
                     .map_err(Error::Gelf)?;
                 tokio::spawn(task);
-                let subscriber = builder.with_max_level(*level).finish().with(layer);
-                Ok((Box::new(subscriber), None))
+                Some(layer)
+            } else {
+                None
             }
-        }
+        } else {
+            None
+        };
+
+        // configure the registry with optional outputs configured above
+        let registry = registry
+            .with(std_out_layer_json)
+            .with(std_out_layer)
+            .with(std_err_layer_json)
+            .with(std_err_layer)
+            .with(file_layer_json)
+            .with(file_layer);
+        #[cfg(feature = "systemd")]
+        let registry = registry.with(journald_layer);
+        #[cfg(feature = "gelf")]
+        let registry = registry.with(gelf_layer);
+
+        // configure the registry subscriber as the global default,
+        // panics if something goes wrong.
+        registry.init();
+
+        let log_msgs = match (self.1, info_msgs.is_empty()) {
```

Comment:
This type should be a struct so that the reader does not need to check the definition to know what `self.1` means.

Reply:
I will now put this into a struct.